### PR TITLE
Enable saving for multitumor nested dichotomous

### PR DESCRIPTION
**Pull request by bishwobhandari**
- added nested dichotomous and multi tumor dataset types to support save analysis in UI. 
- added a mock of execution results for both types

### CODE REVIEW

**Comment by bishwobhandari** [comment]
@shapiromatron failed in unit test because bmds types does not have nested dichotomous yet.

File: tests/analysis/test_validators.py [hunk 1]
```diff
@@ -76,6 +76,242 @@ def test_bmds3_partial(self):
...
+        print(data)
+        assert validators.validate_input(data, partial=True) is None
+
+        # but fails when complete
+        with pytest.raises(ValidationError) as err:
+            validators.validate_input(data)
+        _missing_field(err, "datasets")
+
+        # add datasets, try again
+        data["datasets"] = [
```
Comment (shapiromatron) [inline tests/analysis/test_validators.py:93]: can we make a fixture? using `# fmt: off` w/ formatting like javascript?  see other examples in bmds-`server\tests\conftest.py`

File: bmds_server/main/settings/base.py [hunk 2]
```diff
@@ -79,8 +79,8 @@
     "default": {
         "ENGINE": "django.db.backends.postgresql",
         "NAME": os.getenv("DJANGO_DB_NAME", "bmds-online"),
-        "USER": os.getenv("DJANGO_DB_USER", "bmds-online"),
```
Comment (shapiromatron) [inline bmds_server/main/settings/base.py:82]: remove these lines :). Just create a `bmds_server/main/settings/local.py` and put your custom settings for the database there.

File: frontend/src/constants/dataConstants.js [hunk 3]
```diff
@@ -130,9 +130,9 @@ export const DATA_CONTINUOUS_SUMMARY = "CS",
                         response_name: "Incidence",
                     },
                     doses: ["", "", "", "", ""],
-                    litter_n: ["", "", "", "", ""],
```
Comment (shapiromatron) [inline frontend/src/constants/dataConstants.js:133]: nice, i like the plurality change

File: tests/analysis/test_validators.py [hunk 4]
```diff
@@ -76,6 +76,242 @@ def test_bmds3_partial(self):
         assert validators.validate_input(data, partial=True) is None
         assert validators.validate_input(data) is None
 
+    def test_nested_dichotomous(self):
+        data: Dict[str, Any] = {
+            "bmds_version": bmds.constants.BMDS330,
+            "dataset_type": bmds.constants.NESTED_DICHOTOMOUS,
+        }
+        print(data)
```
Comment (shapiromatron) [inline tests/analysis/test_validators.py:84]: no prints

File: tests/analysis/test_validators.py [hunk 5]
```diff
@@ -76,6 +76,242 @@ def test_bmds3_partial(self):
...
+        assert validators.validate_input(data, partial=True) is None
+
+        # but fails when complete
+        with pytest.raises(ValidationError) as err:
+            validators.validate_input(data)
+        _missing_field(err, "datasets")
+
+        # add datasets, try again
+        data["datasets"] = [
```
Comment (shapiromatron) [inline tests/analysis/test_validators.py:277]: add check that for multitumor, at least two dichotmous datasets exist

**Review by shapiromatron — CHANGES_REQUESTED** [review]
Really nice start. See a few requested changes; let me know if we need to chat about the particulars...

File: tests/analysis/test_validators.py (hunk 1 above)
Comment (bishwobhandari) [reply tests/analysis/test_validators.py]: moved dataset into pytest fixtures.

File: tests/analysis/test_validators.py (hunk 5 above)
Comment (bishwobhandari) [reply tests/analysis/test_validators.py:148]: created another dataset for multi tumor which has min item as 2 for datasets and dataset model options

File: bmds_server/main/settings/base.py (hunk 2 above)
Comment (bishwobhandari) [reply bmds_server/main/settings/base.py]: removed

**Comment by bishwobhandari** [comment]
@shapiromatron i made requested changes. please review where i made change

**Pull request merged**